module/spl/spl-taskq.c

```diff
@@ -723,7 +742,11 @@ taskq_init_ent(taskq_ent_t *t)
 {
 	spin_lock_init(&t->tqent_lock);
 	init_waitqueue_head(&t->tqent_waitq);
+#ifdef HAVE_KERNEL_TIMER_SETUP
+	timer_setup(&t->tqent_timer, NULL, 0);
+#else
 	init_timer(&t->tqent_timer);
+#endif
```

Comment:
We should be able to hide away all this uglyness away in a `spl_timer_setup()` wrapper.